### ISSUE TITLE
TLS: support PKCS#1 certificates

### ISSUE THREAD

**Issue by rursprung** [opensearch-project/security#1573]
**Is your feature request related to a problem? Please describe.**
TLS keys in the PKCS#1 format should be supported as well.

**Describe the solution you'd like**
it should be possible to use also PKCS#1 certificates.

**Describe alternatives you've considered**
most systems also allow generating PKCS#8 certificates or they can be manually converted.
for us it works to generate PKCS#8, however it's not the default of cert-manager.

for those using cert-manager, you can add the following to your `Certificate` k8s custom resource:
```yaml
apiVersion: cert-manager.io/v1
kind: Certificate
# ...
spec:
# ...
  privateKey:
    encoding: PKCS8 # required by OpenSearch Security, it can't handle PKCS1 (which is the default of cert-manager)
```

**Additional context**
* PKCS#1 is the default for certificates generated by cert-manager, see the [`CertificatePrivateKey` specification](https://cert-manager.io/docs/reference/api-docs/#cert-manager.io/v1.CertificatePrivateKey)
* see also #52 (i decided to create a dedicated new ticket for this as it's not limited to Let's Encrypt and the other ticket seems to contain other topics as well)
* e.g. Search Guard supports PKCS#1 as well.

when starting OpenSearch Security 1.2.3.0 with PKCS#1 keys you get the following error in the log:
```
{"type": "server", "timestamp": "2022-01-14T08:18:14,945Z", "level": "ERROR", "component": "o.o.s.s.DefaultSecurityKeyStore", "cluster.name": "test-opensearch", "node.name": "test-opensearch-0", "message": "Your keystore or PEM does not contain a key. If you specified a key password, try removing it. If you did not specify a key password, perhaps you need to if the key is in fact password-protected. Maybe you just confused keys and certificates." }
{"type": "server", "timestamp": "2022-01-14T08:18:15,144Z", "level": "ERROR", "component": "o.o.b.OpenSearchUncaughtExceptionHandler", "cluster.name": "test-opensearch", "node.name": "test-opensearch-0", "message": "uncaught exception in thread [main]", 
"stacktrace": ["org.opensearch.bootstrap.StartupException: java.lang.IllegalStateException: failed to load plugin class [org.opensearch.security.OpenSearchSecurityPlugin]",
"at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:182) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.OpenSearch.execute(OpenSearch.java:169) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:100) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:138) ~[opensearch-cli-1.2.3.jar:1.2.3]",
"at org.opensearch.cli.Command.main(Command.java:101) ~[opensearch-cli-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:135) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:101) ~[opensearch-1.2.3.jar:1.2.3]",
"Caused by: java.lang.IllegalStateException: failed to load plugin class [org.opensearch.security.OpenSearchSecurityPlugin]",
"at org.opensearch.plugins.PluginsService.loadPlugin(PluginsService.java:790) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.plugins.PluginsService.loadBundle(PluginsService.java:726) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.plugins.PluginsService.loadBundles(PluginsService.java:528) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.plugins.PluginsService.<init>(PluginsService.java:194) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.node.Node.<init>(Node.java:396) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.node.Node.<init>(Node.java:319) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:242) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.Bootstrap.setup(Bootstrap.java:242) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:412) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:178) ~[opensearch-1.2.3.jar:1.2.3]",
"... 6 more",
"Caused by: java.lang.reflect.InvocationTargetException",
"at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]",
"at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:?]",
"at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]",
"at java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[?:?]",
"at org.opensearch.plugins.PluginsService.loadPlugin(PluginsService.java:781) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.plugins.PluginsService.loadBundle(PluginsService.java:726) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.plugins.PluginsService.loadBundles(PluginsService.java:528) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.plugins.PluginsService.<init>(PluginsService.java:194) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.node.Node.<init>(Node.java:396) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.node.Node.<init>(Node.java:319) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:242) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.Bootstrap.setup(Bootstrap.java:242) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:412) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:178) ~[opensearch-1.2.3.jar:1.2.3]",
"... 6 more",
"Caused by: org.opensearch.OpenSearchSecurityException: Error while initializing transport SSL layer from PEM: java.lang.IllegalArgumentException: File does not contain valid private key: /opt/opensearch/config/certificates/transport/tls.key",
"at org.opensearch.security.ssl.DefaultSecurityKeyStore.initTransportSSLConfig(DefaultSecurityKeyStore.java:419) ~[?:?]",
"at org.opensearch.security.ssl.DefaultSecurityKeyStore.initSSLConfig(DefaultSecurityKeyStore.java:258) ~[?:?]",
"at org.opensearch.security.ssl.DefaultSecurityKeyStore.<init>(DefaultSecurityKeyStore.java:179) ~[?:?]",
"at org.opensearch.security.ssl.OpenSearchSecuritySSLPlugin.<init>(OpenSearchSecuritySSLPlugin.java:218) ~[?:?]",
"at org.opensearch.security.OpenSearchSecurityPlugin.<init>(OpenSearchSecurityPlugin.java:252) ~[?:?]",
"at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]",
"at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:?]",
"at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]",
"at java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[?:?]",
"at org.opensearch.plugins.PluginsService.loadPlugin(PluginsService.java:781) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.plugins.PluginsService.loadBundle(PluginsService.java:726) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.plugins.PluginsService.loadBundles(PluginsService.java:528) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.plugins.PluginsService.<init>(PluginsService.java:194) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.node.Node.<init>(Node.java:396) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.node.Node.<init>(Node.java:319) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:242) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.Bootstrap.setup(Bootstrap.java:242) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:412) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:178) ~[opensearch-1.2.3.jar:1.2.3]",
"... 6 more",
"Caused by: java.lang.IllegalArgumentException: File does not contain valid private key: /opt/opensearch/config/certificates/transport/tls.key",
"at io.netty.handler.ssl.SslContextBuilder.keyManager(SslContextBuilder.java:386) ~[?:?]",
"at io.netty.handler.ssl.SslContextBuilder.forServer(SslContextBuilder.java:120) ~[?:?]",
"at org.opensearch.security.ssl.DefaultSecurityKeyStore.buildSSLServerContext(DefaultSecurityKeyStore.java:869) ~[?:?]",
"at org.opensearch.security.ssl.DefaultSecurityKeyStore.initTransportSSLConfig(DefaultSecurityKeyStore.java:405) ~[?:?]",
"at org.opensearch.security.ssl.DefaultSecurityKeyStore.initSSLConfig(DefaultSecurityKeyStore.java:258) ~[?:?]",
"at org.opensearch.security.ssl.DefaultSecurityKeyStore.<init>(DefaultSecurityKeyStore.java:179) ~[?:?]",
"at org.opensearch.security.ssl.OpenSearchSecuritySSLPlugin.<init>(OpenSearchSecuritySSLPlugin.java:218) ~[?:?]",
"at org.opensearch.security.OpenSearchSecurityPlugin.<init>(OpenSearchSecurityPlugin.java:252) ~[?:?]",
"at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]",
"at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:?]",
"at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]",
"at java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[?:?]",
"at org.opensearch.plugins.PluginsService.loadPlugin(PluginsService.java:781) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.plugins.PluginsService.loadBundle(PluginsService.java:726) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.plugins.PluginsService.loadBundles(PluginsService.java:528) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.plugins.PluginsService.<init>(PluginsService.java:194) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.node.Node.<init>(Node.java:396) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.node.Node.<init>(Node.java:319) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:242) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.Bootstrap.setup(Bootstrap.java:242) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:412) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:178) ~[opensearch-1.2.3.jar:1.2.3]",
"... 6 more",
"Caused by: java.security.spec.InvalidKeySpecException: Neither RSA, DSA nor EC worked",
"at io.netty.handler.ssl.SslContext.getPrivateKeyFromByteBuffer(SslContext.java:1155) ~[?:?]",
"at io.netty.handler.ssl.SslContext.toPrivateKey(SslContext.java:1123) ~[?:?]",
"at io.netty.handler.ssl.SslContextBuilder.keyManager(SslContextBuilder.java:384) ~[?:?]",
"at io.netty.handler.ssl.SslContextBuilder.forServer(SslContextBuilder.java:120) ~[?:?]",
"at org.opensearch.security.ssl.DefaultSecurityKeyStore.buildSSLServerContext(DefaultSecurityKeyStore.java:869) ~[?:?]",
"at org.opensearch.security.ssl.DefaultSecurityKeyStore.initTransportSSLConfig(DefaultSecurityKeyStore.java:405) ~[?:?]",
"at org.opensearch.security.ssl.DefaultSecurityKeyStore.initSSLConfig(DefaultSecurityKeyStore.java:258) ~[?:?]",
"at org.opensearch.security.ssl.DefaultSecurityKeyStore.<init>(DefaultSecurityKeyStore.java:179) ~[?:?]",
"at org.opensearch.security.ssl.OpenSearchSecuritySSLPlugin.<init>(OpenSearchSecuritySSLPlugin.java:218) ~[?:?]",
"at org.opensearch.security.OpenSearchSecurityPlugin.<init>(OpenSearchSecurityPlugin.java:252) ~[?:?]",
"at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]",
"at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:?]",
"at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]",
"at java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[?:?]",
"at org.opensearch.plugins.PluginsService.loadPlugin(PluginsService.java:781) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.plugins.PluginsService.loadBundle(PluginsService.java:726) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.plugins.PluginsService.loadBundles(PluginsService.java:528) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.plugins.PluginsService.<init>(PluginsService.java:194) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.node.Node.<init>(Node.java:396) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.node.Node.<init>(Node.java:319) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:242) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.Bootstrap.setup(Bootstrap.java:242) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:412) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:178) ~[opensearch-1.2.3.jar:1.2.3]",
"... 6 more",
"Caused by: java.security.spec.InvalidKeySpecException: java.security.InvalidKeyException: IOException : algid parse error, not a sequence",
"at sun.security.ec.ECKeyFactory.engineGeneratePrivate(ECKeyFactory.java:169) ~[?:?]",
"at java.security.KeyFactory.generatePrivate(KeyFactory.java:390) ~[?:?]",
"at io.netty.handler.ssl.SslContext.getPrivateKeyFromByteBuffer(SslContext.java:1153) ~[?:?]",
"at io.netty.handler.ssl.SslContext.toPrivateKey(SslContext.java:1123) ~[?:?]",
"at io.netty.handler.ssl.SslContextBuilder.keyManager(SslContextBuilder.java:384) ~[?:?]",
"at io.netty.handler.ssl.SslContextBuilder.forServer(SslContextBuilder.java:120) ~[?:?]",
"at org.opensearch.security.ssl.DefaultSecurityKeyStore.buildSSLServerContext(DefaultSecurityKeyStore.java:869) ~[?:?]",
"at org.opensearch.security.ssl.DefaultSecurityKeyStore.initTransportSSLConfig(DefaultSecurityKeyStore.java:405) ~[?:?]",
"at org.opensearch.security.ssl.DefaultSecurityKeyStore.initSSLConfig(DefaultSecurityKeyStore.java:258) ~[?:?]",
"at org.opensearch.security.ssl.DefaultSecurityKeyStore.<init>(DefaultSecurityKeyStore.java:179) ~[?:?]",
"at org.opensearch.security.ssl.OpenSearchSecuritySSLPlugin.<init>(OpenSearchSecuritySSLPlugin.java:218) ~[?:?]",
"at org.opensearch.security.OpenSearchSecurityPlugin.<init>(OpenSearchSecurityPlugin.java:252) ~[?:?]",
"at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]",
"at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:?]",
"at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]",
"at java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[?:?]",
"at org.opensearch.plugins.PluginsService.loadPlugin(PluginsService.java:781) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.plugins.PluginsService.loadBundle(PluginsService.java:726) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.plugins.PluginsService.loadBundles(PluginsService.java:528) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.plugins.PluginsService.<init>(PluginsService.java:194) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.node.Node.<init>(Node.java:396) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.node.Node.<init>(Node.java:319) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:242) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.Bootstrap.setup(Bootstrap.java:242) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:412) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:178) ~[opensearch-1.2.3.jar:1.2.3]",
"... 6 more",
"Caused by: java.security.InvalidKeyException: IOException : algid parse error, not a sequence",
"at sun.security.pkcs.PKCS8Key.decode(PKCS8Key.java:350) ~[?:?]",
"at sun.security.pkcs.PKCS8Key.decode(PKCS8Key.java:355) ~[?:?]",
"at sun.security.ec.ECPrivateKeyImpl.<init>(ECPrivateKeyImpl.java:74) ~[?:?]",
"at sun.security.ec.ECKeyFactory.implGeneratePrivate(ECKeyFactory.java:237) ~[?:?]",
"at sun.security.ec.ECKeyFactory.engineGeneratePrivate(ECKeyFactory.java:165) ~[?:?]",
"at java.security.KeyFactory.generatePrivate(KeyFactory.java:390) ~[?:?]",
"at io.netty.handler.ssl.SslContext.getPrivateKeyFromByteBuffer(SslContext.java:1153) ~[?:?]",
"at io.netty.handler.ssl.SslContext.toPrivateKey(SslContext.java:1123) ~[?:?]",
"at io.netty.handler.ssl.SslContextBuilder.keyManager(SslContextBuilder.java:384) ~[?:?]",
"at io.netty.handler.ssl.SslContextBuilder.forServer(SslContextBuilder.java:120) ~[?:?]",
"at org.opensearch.security.ssl.DefaultSecurityKeyStore.buildSSLServerContext(DefaultSecurityKeyStore.java:869) ~[?:?]",
"at org.opensearch.security.ssl.DefaultSecurityKeyStore.initTransportSSLConfig(DefaultSecurityKeyStore.java:405) ~[?:?]",
"at org.opensearch.security.ssl.DefaultSecurityKeyStore.initSSLConfig(DefaultSecurityKeyStore.java:258) ~[?:?]",
"at org.opensearch.security.ssl.DefaultSecurityKeyStore.<init>(DefaultSecurityKeyStore.java:179) ~[?:?]",
"at org.opensearch.security.ssl.OpenSearchSecuritySSLPlugin.<init>(OpenSearchSecuritySSLPlugin.java:218) ~[?:?]",
"at org.opensearch.security.OpenSearchSecurityPlugin.<init>(OpenSearchSecurityPlugin.java:252) ~[?:?]",
"at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]",
"at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:?]",
"at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]",
"at java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[?:?]",
"at org.opensearch.plugins.PluginsService.loadPlugin(PluginsService.java:781) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.plugins.PluginsService.loadBundle(PluginsService.java:726) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.plugins.PluginsService.loadBundles(PluginsService.java:528) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.plugins.PluginsService.<init>(PluginsService.java:194) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.node.Node.<init>(Node.java:396) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.node.Node.<init>(Node.java:319) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:242) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.Bootstrap.setup(Bootstrap.java:242) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:412) ~[opensearch-1.2.3.jar:1.2.3]",
"at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:178) ~[opensearch-1.2.3.jar:1.2.3]",
"... 6 more"] }
uncaught exception in thread [main]
java.lang.IllegalStateException: failed to load plugin class [org.opensearch.security.OpenSearchSecurityPlugin]
Likely root cause: java.security.InvalidKeyException: IOException : algid parse error, not a sequence
	at java.base/sun.security.pkcs.PKCS8Key.decode(PKCS8Key.java:350)
	at java.base/sun.security.pkcs.PKCS8Key.decode(PKCS8Key.java:355)
	at jdk.crypto.ec/sun.security.ec.ECPrivateKeyImpl.<init>(ECPrivateKeyImpl.java:74)
	at jdk.crypto.ec/sun.security.ec.ECKeyFactory.implGeneratePrivate(ECKeyFactory.java:237)
	at jdk.crypto.ec/sun.security.ec.ECKeyFactory.engineGeneratePrivate(ECKeyFactory.java:165)
	at java.base/java.security.KeyFactory.generatePrivate(KeyFactory.java:390)
	at io.netty.handler.ssl.SslContext.getPrivateKeyFromByteBuffer(SslContext.java:1153)
	at io.netty.handler.ssl.SslContext.toPrivateKey(SslContext.java:1123)
	at io.netty.handler.ssl.SslContextBuilder.keyManager(SslContextBuilder.java:384)
	at io.netty.handler.ssl.SslContextBuilder.forServer(SslContextBuilder.java:120)
	at org.opensearch.security.ssl.DefaultSecurityKeyStore.buildSSLServerContext(DefaultSecurityKeyStore.java:869)
	at org.opensearch.security.ssl.DefaultSecurityKeyStore.initTransportSSLConfig(DefaultSecurityKeyStore.java:405)
	at org.opensearch.security.ssl.DefaultSecurityKeyStore.initSSLConfig(DefaultSecurityKeyStore.java:258)
	at org.opensearch.security.ssl.DefaultSecurityKeyStore.<init>(DefaultSecurityKeyStore.java:179)
	at org.opensearch.security.ssl.OpenSearchSecuritySSLPlugin.<init>(OpenSearchSecuritySSLPlugin.java:218)
	at org.opensearch.security.OpenSearchSecurityPlugin.<init>(OpenSearchSecurityPlugin.java:252)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at org.opensearch.plugins.PluginsService.loadPlugin(PluginsService.java:781)
	at org.opensearch.plugins.PluginsService.loadBundle(PluginsService.java:726)
	at org.opensearch.plugins.PluginsService.loadBundles(PluginsService.java:528)
	at org.opensearch.plugins.PluginsService.<init>(PluginsService.java:194)
	at org.opensearch.node.Node.<init>(Node.java:396)
	at org.opensearch.node.Node.<init>(Node.java:319)
	at org.opensearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:242)
	at org.opensearch.bootstrap.Bootstrap.setup(Bootstrap.java:242)
	at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:412)
	at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:178)
	<<<truncated>>>
```

background on the error message: https://stackoverflow.com/questions/31941413/invalidkeyspecexception-algid-parse-error-not-a-sequence

**Comment by rursprung** [comment]
by pure chance i stumbled over this the fact that java 17 now contains PKCS#1 support in `javax.crypto`: https://bugs.openjdk.java.net/browse/JDK-8023980

i'm not familiar with the security code of opensearch (or any code of opensearch, really :D), but if you're using bouncy castle then PKCS#1 support should already be possible (as BC supports it), if you're using pure `javax.crypto` for this then it could only be supported once java 17 is the lowest supported JDK.

**Comment by cliu123** [comment]
The issue is being investigated. 
I see the issue is being labeled for 2.2.0 release. Is this needed for the particular release? If not, could the label be removed?

**Comment by peternied** [comment]
@cwperks Last we discussed this issue it sounds like you were going to attempt to fix in bouncy castle, could you update this issue with what you are thinking?

**Comment by cwperks** [comment]
Hey @peternied, this is a work in progress and current plans are to open a PR with netty to add support for pkcs#1 in the SSLContextBuilder. Netty currently only supports PKCS#8.

**Comment by cwperks** [comment]
Yes, you can remove the label. We can schedule this for a release once the netty changes have been complete and released in a future version of netty.

**Comment by cwperks** [comment]
Submitted a PR to netty to add support for PKCS#1 keys. PR is [here](https://github.com/netty/netty/pull/12670).

**Comment by cwperks** [comment]
The netty [PR](https://github.com/netty/netty/pull/12670) has been merged as of this morning and is tagged in the milestone for `4.1.80.Final`. Once the netty release is issued we can schedule this change to be completed.

**Comment by cwperks** [comment]
Netty released `4.1.80.Final` this past Friday: https://netty.io/news/2022/08/26/4-1-80-Final.html

We can now close the loop on this.

**Comment by rursprung** [comment]
> Netty released `4.1.80.Final` this past Friday: https://netty.io/news/2022/08/26/4-1-80-Final.html
> 
> We can now close the loop on this.

just saw https://github.com/netty/netty/issues/12745 - maybe you need to wait for this one to be released as well as it was created after the 4.1.80.Final release?

**Comment by cwperks** [comment]
Netty released https://netty.io/news/2022/09/08/4-1-81-Final.html with 2 bugfixes related to the BouncyCastlePemReader changes. This can now be upgraded in core and completed.
